templates/git_client_side_hooks/pre-push

```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+make golangci
+MAKE_GOLANGCI=$?
+
+make verify
+MAKE_VERIFY=$?
+
+if [ $MAKE_GOLANGCI != 0 ]
+then
+   echo "The command: make golangci failed! Please fix and try again!"
+   exit 1
+fi
+
+if [ $MAKE_VERIFY != 0 ]
+then
+   echo "The command: make verify failed! Please fix and try again!"
+   exit 1
+fi
```

Comment:
missing new line.

Reply:
Could you check the comment here: https://github.com/antrea-io/antrea/issues/3611#issuecomment-1096069105,
I think it would better to provide a make target like `make copy-git-template` to copy this new pre-push script to user's local .git folder. and please update the `CONTRIBUTING.md` doc about this change, so new comer can learn and use it.
Could you enhance the pre-push script to run `make golangci` or `make verify` based on `git diff`?

Reply:
Why do we need `git diff` here? If there is nothing to push then git returns `Everything up-to-date` by default.

Reply:
> Could you check the comment here: [#3611 (comment)](https://github.com/antrea-io/antrea/issues/3611#issuecomment-1096069105), I think it would better to provide a make target like `make copy-git-template` to copy this new pre-push script to user's local .git folder. and please update the `CONTRIBUTING.md` doc about this change, so new comer can learn and use it. Could you enhance the pre-push script to run `make golangci` or `make verify` based on `git diff`?

You mean a script which will override the default hook with this one in the user's local git folder?

Reply:
the `git diff` can help to skip `make golangci` if there is no golang code change and can also skip `make verify` if there is no doc related change. this can help to save the time before a real push.
and yes, I mean a script or some steps in Makefile directly.

Reply:
I saw you added a description to let developer where is path of pre-defined pre-push hook, but I think you can provide a new target in (Makefile)[https://github.com/antrea-io/antrea/blob/5905e98878b91cdf36a7288859a33ee6277f3b8a/Makefile] to copy pre-defined hooks automatically via `make copy-hooks`, the script should be similar like below:
```
.PHONY: copy-hooks
copy-hooks:
	@echo "===> Copying Antrea git hooks to local <==="
	cp templates/git_client_side_hooks/pre-push .git/hooks/
```

Reply:
btw, Could you squash all three commits into one? thanks.